pagefaultgames/pokerogue
# [Bug] Added fix for 3722 so that text correctly appears when sleep fails in electric terrain

This fix is for this issue:
https://github.com/pagefaultgames/pokerogue/issues/3722

## What are the changes the user will see?
The user will now, when attempting to put a pokemon to sleep on electric terrain, will receive a message why it failed

## Why am I making these changes?

I needed to add fixes to drowsy (from yawn) as well as to the tryStatusEffect 



Makes it more similar to the mainline games and pokemon showdown

## What are the changes from a developer perspective?

A simple change was added in the drowsy state check so that the message will appear.  I also had to add logic to the final


### Screenshots/Videos
See below

## How to test the changes?

I used override, I honestly am not sure how best to test yet.  I set my pokemon to Tapu Koko with spore and yawn as moves and the opponent as well to test.   Below is the basic override to see things. 
```
  readonly STARTER_SPECIES_OVERRIDE: Species | number = Species.TAPU_KOKO;
  readonly ABILITY_OVERRIDE: Abilities = Abilities.NONE;
  readonly PASSIVE_ABILITY_OVERRIDE: Abilities = Abilities.NONE;
  readonly STATUS_OVERRIDE: StatusEffect = StatusEffect.NONE;
  readonly GENDER_OVERRIDE: Gender | null = null;
  readonly MOVESET_OVERRIDE: Array<Moves> = [Moves.SPORE, Moves.YAWN];
  readonly SHINY_OVERRIDE: boolean = false;
  readonly VARIANT_OVERRIDE: Variant = 0;
```




## Checklist
- I set the target branch to beta
- Not sure how best to look at this but I confirmed in the issue that I would fix it
- Yes
- Yes
- I considered it but decided no for now
- Text was already there.  Just no triggering logic
-  Yes
    - No?  But I honestly am not sure how my code might be causing issues.  
    

> FAIL  |main| src/test/moves/flower_shield.test.ts > Moves - Flower Shield > does not increase defense of a pokemon in semi-vulnerable state
> AssertionError: expected +0 to be 1 // Object.is equality
> 
> - Expected
> + Received
> 
> - 1
> + 0
> 

  - Yes

Original Issue on display

https://github.com/user-attachments/assets/9658fa79-3638-4274-9a0a-b6f48862a5ec

Fixes both with spore and sing to showcase 


https://github.com/user-attachments/assets/92bb4ba5-5a34-4019-8069-9d0a88a61027


https://github.com/user-attachments/assets/044916ef-c235-4aaa-b343-54e8c079406c
